### DISCUSSION TITLE
[Core Protocol] Run fault proof system locally

### DISCUSSION THREAD

**Discussion by GCdePaula** [ethereum-optimism/developers#328] [Core Protocol]
### Did you check the documentation?

- [X] Yes

### Did you read the specs?

- [X] Yes
- [ ] No

### Did you check for duplicate questions?

- [X] Yes

### Issue Description

I'm trying to experiment with the fault proof system locally on devnet. My goal is testing and playing around with the fault proofs. To that end, I wanted to run a dispute game locally from start to finish and interact with it.

I saw there are scripts to interact and visualize a dispute game. But I'm having some trouble before reaching that point.

I'm following the instructions to run `op-challenger` on this [readme](https://github.com/ethereum-optimism/optimism/tree/v1.7.2/op-challenger), at tag `v1.7.2`.

I'm getting the following error:

```
INFO [03-30|11:43:56.293] Starting op-challenger                   version=v0.1.0-99a53381-1710933867-dev
CRIT [03-30|11:43:56.293] Application failed                       err="failed to setup: flag l1-beacon is required"
```

These are the steps I'm doing, after cloning and checking out `v1.7.2`:

```
#!/usr/bin/env bash

cd op-challenger
make clean
make
cd ..

make devnet-clean
make devnet-up
make op-challenger

export DISPUTE_GAME_FACTORY=$(jq -r .DisputeGameFactoryProxy .devnet/addresses.json)

./op-challenger/bin/op-challenger \
  --trace-type cannon \
  --l1-eth-rpc http://localhost:8545 \
  --rollup-rpc http://localhost:9546 \
  --game-factory-address $DISPUTE_GAME_FACTORY \
  --datadir temp/challenger-data \
  --cannon-rollup-config .devnet/rollup.json  \
  --cannon-l2-genesis .devnet/genesis-l2.json \
  --cannon-bin ./cannon/bin/cannon \
  --cannon-server ./op-program/bin/op-program \
  --cannon-prestate ./op-program/bin/prestate.json \
  --cannon-l2 http://localhost:9545 \
  --mnemonic "test test test test test test test test test test test junk" \
  --hd-path "m/44'/60'/0'/0/8" \
  --num-confirmations 1
```

Any insights? It looks like I'm doing something silly.

Also, after this, how do I configure the behavior of this local chain? Like shortening challenge windows, etc. I see the jsons at `.devnet/`, but I'm lost. Can you point me to any documentation?

**Comment by Chomtana** [comment]
You need to supply the L1 beacon node to op-challenger

```
--l1-beacon <BeaconNodeEndpoint>
```

You can use Quicknode for this

**Reply by GCdePaula** [reply]
Even on `devnet`? Reiterating, I don't want to run a `mainnet` or `testnet` node.

Like, I wanted to experiment locally on my computer with the fault-proof system.

**Reply by Chomtana** [reply]
Maybe there is a discussion on too much beacon node requirement

https://github.com/ethereum-optimism/developers/discussions/343
